File: scratch/find_splits.js
const fs = require('fs');
const path = require('path');

function walkDir(dir, callback) {
    fs.readdirSync(dir).forEach(f => {
        let dirPath = path.join(dir, f);
        let isDirectory = fs.statSync(dirPath).isDirectory();
        if (isDirectory) {
            if (f !== 'node_modules' && f !== '.git' && f !== 'brain' && f !== '.antigravity') {
                walkDir(dirPath, callback);
            }
        } else {
            callback(dirPath);
        }
    });
}

const baseDir = '.';
const pattern = /\.split\(['"]\s*['"]\)/;

walkDir(baseDir, (filePath) => {
    if (filePath.endsWith('.js') || filePath.endsWith('.html')) {
        const content = fs.readFileSync(filePath, 'utf8');
        const lines = content.split('\n');
        lines.forEach((line, idx) => {
            if (pattern.test(line)) {
                console.log(`${filePath}:${idx + 1}: ${line.trim()}`);
            }
        });
    }
});

File: scratch/find_splits.py
import os
import re

pattern = re.compile(r"\.split\(['\"]\s*['\"]\)")

def search_files(dir_path):
    for root, dirs, files in os.walk(dir_path):
        # Exclude directories
        dirs[:] = [d for d in dirs if d not in ('node_modules', '.git', 'brain', '.antigravity')]
        for file in files:
            if file.endswith('.js') or file.endswith('.html'):
                file_path = os.path.join(root, file)
                try:
                    with open(file_path, 'r', encoding='utf-8') as f:
                        for idx, line in enumerate(f):
                            if pattern.search(line):
                                print(f"{file_path}:{idx + 1}: {line.strip()}")
                except Exception as e:
                    pass

search_files('.')
